ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantState.scala

```diff
@@ -23,44 +23,32 @@ import com.daml.ledger.participant.state.v2.{
 import com.daml.lf.data.{Ref, Time}
 import com.daml.lf.transaction.SubmittedTransaction
 import com.daml.logging.LoggingContext
-import com.daml.metrics.Metrics
 import com.daml.telemetry.TelemetryContext
 
 /** Implements read and write operations required for running a participant server.
   *
-  * Adapts [[LedgerReader]] and [[LedgerWriter]] interfaces to [[com.daml.ledger.participant.state.v2.ReadService]] and
-  * [[com.daml.ledger.participant.state.v2.WriteService]], respectively.
+  * Unifies [[KeyValueParticipantStateReader]] and [[KeyValueParticipantStateWriter]] interfaces
```

Comment:
The point of this class has been to adapt the KV ledger reader/writer interfaces to the `ReadService`/`WriteService` abstractions. You are changing it to essentially act as a `ReadWriteService` implementation, delegating from a `ReadService` and `WriteService` instance.
If we need the latter, we must still keep the adapting aspect, e.g., by introducing a factory method that creates the reader/writer adapters. Otherwise, usage of the simplified KV APIs suffers as you cannot simply adapt a `LedgerReader`/`LedgerWriter` to the abstractions used internally by the participant server.

Reply:
That's a good point.
Thinking of it, we actually removed the usage of `readWriteServiceOwner` from the `Runner`, and I think it's only used in some tests. 
I suggest 2 alternatives:
- we remove the interface all together
- rollback the changes here, doable without a lot of pain as well
I'm more in favor of the first one, so that we keep the ReadService and WriteService split as much as possible.

Reply:
Can you clarify what you mean by 'interface' here, please?

Reply:
Ah yes, the `interface` naming makes no sense. I was referring to the `KeyValueParticipantState`